aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -70,10 +69,13 @@
   private static final Cache<Key, SignedComponent> SIGNED_COMPONENT_CACHE =
       Caffeine.newBuilder().expireAfterWrite(30, TimeUnit.SECONDS).maximumSize(100).build();
 
-  private static final ScheduledExecutorService TOKEN_REFRESH_EXECUTOR =
-      ThreadPools.newScheduledPool("s3-signer-token-refresh", 1);
   private static final String SCOPE = "sign";
-  private static final AtomicReference<RESTClient> HTTP_CLIENT_REF = new AtomicReference<>();
+
+  @SuppressWarnings("immutables:incompat")
```

Comment:
this is to suppress `warning: (immutables:incompat) Avoid introduction of fields (except constants) in abstract value types`

Reply:
This is static. Why would it change the interface?

Reply:
this seems to be an intended warning (see https://github.com/immutables/immutables/issues/182 for some additional context) because the field isn't static final and thus can be modified (which was the reason for bb95e8bb1a9c283b86afeae978d2f838456c5d03).